deployer/__main__.py

```diff
@@ -18,6 +18,35 @@
 yaml = YAML(typ="safe", pure=True)
 
 
+def setup_cluster_credentials(cluster_name):
+    """
+    Quickly gain command-line access to a cluster by updating the current
+    kubeconfig file to include the deployer's access credentials for the named
+    cluster and mark it as the cluster to work against by default.
+
+    This function is to be used with the `setup-cluster-credentials` CLI
+    command only - it is not used by the rest of the deployer codebase.
+    """
+
+    # Validate our config with JSON Schema first before continuing
+    validate(cluster_name)
+
+    config_file_path = (
+        Path(os.getcwd()) / "config/clusters" / f"{cluster_name}.cluster.yaml"
+    )
+    with open(config_file_path) as f:
+        cluster = Cluster(yaml.load(f))
+
+    # Cluster.auth() method has the context manager decorator so cannot call
+    # it like a normal function
+    with cluster.auth():
+        # This command will spawn a new shell with all the env vars (including
+        # KUBECONFIG) inherited, and once you quit that shell the python program
+        # will resume as usual.
+        subprocess.check_call([os.environ["SHELL"], "-l"])
+        return None
```

Comment:
why explicitly `return None` here? Isn't that default python behavior?

Reply:
Yeah, it can probably be removed now we have some actual code in this block. But when I first wrote this, I realised that `cluster.auth()` is a context manager so I needed to wrap _something_ and couldn't just leave it blank.

Reply:
Done in [`7152233` (#977)](https://github.com/2i2c-org/infrastructure/pull/977/commits/715223389e7c13192a1eecb92a0cb5349ffd82d5)